pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -654,8 +663,13 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnable
 	}
 
 	// Otherwise has ambient enabled. Note: this is a synthetic label, not existing in the real Pod.
-	if b.push.SupportsTunnel(e.Network, e.Address) {
-		supportsTunnel = true
+	// Should all ip addresses need to support tunneling ?
```

Comment:
Not sure, but this is definitely not right, since we are just checking if the *last* one does

Reply:
Okay, thanks for the information, I define a `lastAddr`, and following the existing logic to check the last address of endpoint.

Reply:
Hi @howardjohn, okay, thanks for the information. I define the variable of `lastAddr`, then following the existing logic to check the last address of endpoint for tunnel supporting.

Reply:
Hi @howardjohn , okay, thanks for the information, I define the variable of `supportTunnelAddr`, then find the correct address which can support tunnel and assign it to `supportTunnelAddr`. This may be more reasonable.

Reply:
Hi @howardjohn , okay, thanks for the information, I define the variable of `supportTunnelAddr`, then find the correct address which can support tunnel and assign it to `supportTunnelAddr`. This may be more reasonable.  Any other idea?

Reply:
Hi @howardjohn , okay, thanks for the information, I define the variable of `supportTunnelAddr`, then find the correct address which can support tunnel and assign it to `supportTunnelAddr`. This may be more reasonable.